### PR TITLE
Add admin view to generate codes

### PR DESCRIPTION
**Pull request by maxpeterson**

### CODE REVIEW

**Comment by maxpeterson** [comment]
@meshy want to review?

File: profiles/modules/codes/templates/admin/profiles/code/change_list.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "admin/change_list.html" %}
+
+{% load url from future %}
+{% load i18n %}
+
+{% block content %}{{ block.super }}
+    <form action="{% url "admin:code_auto_generate" %}" method="POST" enctype="multipart/form-data">
```
Comment (meshy) [inline profiles/modules/codes/templates/admin/profiles/code/change_list.html]: Why is this multipart?

File: profiles/modules/codes/admin.py [hunk 2]
```diff
@@ -1,12 +1,72 @@
...
+        print 'auto_generate'
+        if request.method == 'POST':
+            if not self.has_add_permission(request):
+                raise PermissionDenied
+            form = AutoGenerateForm(request.POST)
+            if form.is_valid():
+                number = form.cleaned_data.get('number')
+                length = form.cleaned_data.get('length')
+                for i in range(number):
+                    code = ''.join(random.choice(string.ascii_uppercase + string.digits) for x in range(length))
```
Comment (meshy) [inline profiles/modules/codes/admin.py]: What does this do?

File: profiles/modules/codes/admin.py [hunk 3]
```diff
@@ -1,12 +1,72 @@
...
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseRedirect
+from django.utils.translation import ugettext_lazy as _
 
 from .models import Code
 
 
+class AutoGenerateForm(forms.Form):
+    number = forms.IntegerField(label=_('Number of codes to generate'))
+    length = forms.IntegerField(
```
Comment (meshy) [inline profiles/modules/codes/admin.py:17]: Should we not dictate length? I imagine -- `length = 1` could be unhelpful... Perhaps a `min_length` on the integer field?

File: profiles/modules/codes/admin.py [hunk 4]
```diff
@@ -1,12 +1,72 @@
...
+        if request.method == 'POST':
+            if not self.has_add_permission(request):
+                raise PermissionDenied
+            form = AutoGenerateForm(request.POST)
+            if form.is_valid():
+                number = form.cleaned_data.get('number')
+                length = form.cleaned_data.get('length')
+                for i in range(number):
+                    code = ''.join(random.choice(string.ascii_uppercase + string.digits) for x in range(length))
+                    Code.objects.create(code=code, is_active=True)
```
Comment (meshy) [inline profiles/modules/codes/admin.py:70]: Are codes unique? What happens if there's more than one the same?

Comment (maxpeterson) [inline profiles/modules/codes/admin.py:70]: Good point. They should be.

File: profiles/modules/codes/admin.py (hunk 3 above)
Comment (meshy) [inline profiles/modules/codes/admin.py:17]: oops `CharField`, not `IntegerField`.

**Pull request merged**